py/server/tests/test_arrow.py

```diff
@@ -0,0 +1,98 @@
+    string_col, datetime_col
+from deephaven.table import Table
+from tests.testbase import BaseTestCase
+
+
+class ArrowTestCase(BaseTestCase):
+    test_table: Table
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cols = [
+            # bool_col(name="Boolean", data=[True, False]),
+            byte_col(name="Byte", data=(1, -1)),
+            char_col(name="Char", data='-1'),
+            short_col(name="Short", data=[1, -1]),
+            int_col(name="Int", data=[1, -1]),
+            long_col(name="Long", data=[1, -1]),
+            long_col(name="NPLong", data=np.array([1, -1], dtype=np.int8)),
+            float_col(name="Float", data=[1.01, -1.01]),
+            double_col(name="Double", data=[1.01, -1.01]),
+            string_col(name="String", data=["foo", "bar"]),
+            datetime_col(name="Datetime", data=[dtypes.DateTime(1), dtypes.DateTime(-1)]),
+        ]
+        cls.test_table = new_table(cols=cols)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        del cls.test_table
+
+    def test_round_trip(self):
+        arrow_table = papq.read_table("tests/data/day_trades.parquet")
+
+        dh_table = dharrow.to_table(arrow_table)
+        pa_table = dharrow.to_arrow(dh_table)
+        dh_table_rt = dharrow.to_table(pa_table)
+        self.assert_table_equals(dh_table, dh_table_rt)
+
+    def test_round_trip_types(self):
+        pa_table = dharrow.to_arrow(self.test_table)
+        dh_table_rt = dharrow.to_table(pa_table)
+        pa_table_rt = dharrow.to_arrow(dh_table_rt)
+        self.assert_table_equals(self.test_table, dh_table_rt)
+        self.assertTrue(pa_table_rt, pa_table)
+
+    def test_round_trip_empty(self):
+        cols = [
+            byte_col(name="Byte", data=()),
+            char_col(name="Char", data=''),
+            short_col(name="Short", data=[]),
+            int_col(name="Int", data=[]),
+            long_col(name="Long", data=[]),
+            long_col(name="NPLong", data=np.array([], dtype=np.int8)),
+            float_col(name="Float", data=[]),
+            double_col(name="Double", data=[]),
+            string_col(name="String", data=[]),
+            datetime_col(name="Datetime", data=[]),
+        ]
+        dh_table = new_table(cols=cols)
+        pa_table = dharrow.to_arrow(dh_table)
+        dh_table_rt = dharrow.to_table(pa_table)
+        self.assert_table_equals(dh_table, dh_table_rt)
+
+    def test_round_trip_cols(self):
+        cols = ["Byte", "Short", "Long", "String", "Datetime"]
+        pa_table = dharrow.to_arrow(self.test_table)
+        pa_table_cols = dharrow.to_arrow(self.test_table, cols=cols)
+        dh_table = dharrow.to_table(pa_table, cols=cols)
+        dh_table_1 = dharrow.to_table(pa_table_cols)
+        self.assert_table_equals(dh_table_1, dh_table)
+
+    def test_crypto_data(self):
+        arrow_table = papq.read_table("tests/data/crypto_trades_1000.parquet")
+
+        with self.assertRaises(DHError) as cm:
```

Comment:
Hmm, is there a reason we want to expect this? Is the issue that the file has boolean columns or some other type that is mismatching somehow?

Reply:
this is just a placeholder for a potential bug we ran into, changed the name to make it more obvious.